library/src/main/java/com/datatorrent/lib/simple/SingleInputMultiOutput.java

```diff
@@ -0,0 +1,94 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package com.datatorrent.lib.simple;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import com.datatorrent.api.DefaultInputPort;
+import com.datatorrent.api.DefaultOutputPort;
+import com.datatorrent.api.annotation.Name;
+import com.datatorrent.api.annotation.Stateless;
+import com.datatorrent.common.util.BaseOperator;
+
+/**
+ * A base implementation of an operator that abstracts away the input and output ports.
+ *
+ * Subclasses should provide the implementation to process a tuple of type I and return a tuple of
+ * type O such that the return value will emit to each output port in the array.
+ *
+ * <p>
+ * <b>Input Port :</b><br>
+ * <b> input :</b> default input port. <br>
+ * <br>
+ * <b>Output Port(s) :</b><br>
+ * <b> output[] :</b> output port list that defaults to two ports. <br>
+ * <br>
+ * <b>Stateful : No</b>, all state is handled through the implementing class. <br>
+ * <b>Partitions : Yes</b>, no dependency among input tuples. <br>
+ * <br>
+ * @displayName Single Input With Multiple Output
+ * @category Simple Operators
+ * @tags simple, single input, multiple output
+ * @param <I> type being received from the input port
+ * @param <O> type being sent from the output port
+ * @since 3.3.0
+ */
+@Stateless
+@Name("single-input-multi-output")
+public abstract class SingleInputMultiOutput<I, O> extends BaseOperator
+{
+  protected static final int DEFAULT_NUM_OUTPUTS = 2;
+
+  protected transient List<DefaultOutputPort<O>> outputs = null;
```

Comment:
Can you please check if this works with the platform?
I think the the List object of OutputPort won't be identified by operator.

Please check with a running apex application whether the ports are identified.